nupic/utils.py

```diff
@@ -127,3 +127,258 @@ def write(self, proto):
     proto.windowSize = self.windowSize
     proto.slidingWindow = self.slidingWindow
     proto.total = self.total
+
+
+
+#######################################################
+# code from functools.lru_cache for Python 3.3, backports from ActiveState:
```

Comment:
Yes, because `https://code.activestate.com/recipes/578078/` states the content is MIT.